src/mds/FSMap.cc

```diff
@@ -477,6 +477,9 @@ Filesystem::ref FSMap::create_filesystem(std::string_view name,
     fs->mds_map.failed.insert(mds_rank_t(0));
 
     fs->mds_map.set_flag(CEPH_MDSMAP_NOT_JOINABLE);
+    if (!(fs->mds_map.test_flag(CEPH_MDSMAP_REFUSE_CLIENT_SESSION))) {
+      fs->mds_map.set_flag(CEPH_MDSMAP_REFUSE_CLIENT_SESSION);
+    }
```

Comment:
@ajarr  @batrick added the feature. 

Admin can set `refuse_client_session` before doing `ceph fs new --recover` and that would then the flag which was already set. That's why I put this `if` block to test it first. I hope it makes sense else I can change it too.

Reply:
@batrick probably meant to add the `--refuse_client_session` flag to the `ceph fs new --recover` command-line to act as a single-shot flag which gets seen by the entire cluster for the recovery phase and gets discarded upon completion. This wouldn't require separate setting and resetting operations for the flag.

Reply:
yeah makes sense. let me correct it.

Reply:
removed the commit from PR, after understanding thoroughly, i think this would need a testcase as well documentation for the same. I'll create a tracker for this and push code for it once this PR is merged.